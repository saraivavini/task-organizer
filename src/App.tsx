import React, { useEffect } from 'react';
import { NativeBaseProvider } from 'native-base';
import { Routes } from './routes';
import { ThemeProvider } from 'styled-components/native';
import './locales';
import { theme } from './global/theme';
import OneSignal from 'react-native-onesignal';
import { ONESIGNAL_APP_ID } from '@env';

OneSignal.setAppId(ONESIGNAL_APP_ID as string);

export default function App() {
  return (
    <NativeBaseProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </NativeBaseProvider>
  );
}
