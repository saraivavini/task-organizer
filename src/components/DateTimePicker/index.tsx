import RNDateTimePicker, {
  DateTimePickerEvent,
} from '@react-native-community/datetimepicker';
import { Box, Pressable, Text, Modal } from 'native-base';
import { useState } from 'react';
import { Platform } from 'react-native';
import { DateHandler } from '../../helpers';
import { Input } from '../Input';

type DateTimePickerProps = {
  type?: 'date' | 'time';
  label?: string | null;
  onChange?: (date: Date) => void;
};

const formatByTypeMapping = {
  date: 'DAY_MONTH_YEAR',
  time: 'HOUR_WITH_MINUTES',
} as const;

const iconByTypeMapping = {
  date: 'calendar',
  time: 'clock',
} as const;

export const DateTimePicker = (props: DateTimePickerProps) => {
  const { type = 'date', label, onChange = () => null } = props;

  const [date, setDate] = useState(new Date());
  const [isVisible, setIsVisible] = useState(false);

  const formattedDate = DateHandler.formatDate(date, formatByTypeMapping[type]);

  const handleChange = (_: DateTimePickerEvent, newDate?: Date) => {
    if (newDate) {
      setDate(newDate);

      setIsVisible(Platform.OS === 'ios');
    }
  };

  const toggleVisible = () => {
    setIsVisible(!isVisible);
    onChange(date);
  };

  const display = Platform.OS === 'ios' ? 'spinner' : 'default';

  return (
    <Box>
      <Input
        label={label}
        isReadOnly
        value={formattedDate}
        icon={iconByTypeMapping[type]}
        onPress={toggleVisible}
      />
      <Modal isOpen={isVisible} onClose={toggleVisible}>
        <Modal.Content>
          <RNDateTimePicker
            display={display}
            value={date}
            mode={type}
            onChange={handleChange}
            is24Hour
          />
        </Modal.Content>
      </Modal>
    </Box>
  );
};
