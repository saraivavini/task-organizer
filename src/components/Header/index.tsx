import { Box } from 'native-base';
import { Logo } from '../Logo';

export const Header = () => {
  return (
    <Box mb={6}>
      <Logo size="small" />
    </Box>
  );
};
