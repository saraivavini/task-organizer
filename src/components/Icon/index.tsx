import { Icon as NBIcon, IIconProps as NBIconProps } from 'native-base';
import { FeatherIcon } from '../../@types';
import Feather from '@expo/vector-icons/Feather';

type IconProps = {
  testID?: string;
  name: FeatherIcon;
} & Pick<NBIconProps, 'color' | 'size'>;

export const Icon = (props: IconProps) => {
  const { name, size, color = 'muted.400', testID } = props;

  return (
    <NBIcon
      as={Feather}
      testID={testID}
      color={color}
      name={name}
      size={size}
    />
  );
};
