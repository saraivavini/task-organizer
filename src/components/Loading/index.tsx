import { Modal, Spinner } from 'native-base';

type LoadingProps = {
  isVisible: boolean;
};

export const Loading = ({ isVisible }: LoadingProps) => {
  return (
    <Modal isOpen={isVisible}>
      <Spinner size="lg" colorScheme={'brand'} />
    </Modal>
  );
};
