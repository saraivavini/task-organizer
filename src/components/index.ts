export * from './Button';
export * from './DateTimePicker';
export * from './Header';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Logo';
export * from './ScreenContainer';
export * from './Loading';
