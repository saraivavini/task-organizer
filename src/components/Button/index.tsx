import { Button as NBButton, Text, Icon as NBIcon } from 'native-base';
import { FeatherIcon } from '../../@types';
import { Icon } from '../Icon';

export type ButtonProps = {
  icon?: FeatherIcon;
  label: string;
  onPress: () => void;
  variant?: 'solid' | 'link';
  testID?: string;
};

export const Button = (props: ButtonProps) => {
  const { label, onPress, icon, variant = 'solid', testID } = props;

  return (
    <NBButton
      testID={testID || 'button'}
      colorScheme="brand"
      size="lg"
      borderRadius="8px"
      onPress={onPress}
      leftIcon={icon ? <Icon size={6} name={icon} /> : undefined}
      variant={variant}
      _text={{
        fontSize: 'lg',
        fontWeight: 'medium',
      }}
    >
      {label}
    </NBButton>
  );
};
