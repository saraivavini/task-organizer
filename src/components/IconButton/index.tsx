import { Pressable } from 'native-base';
import { IColors } from 'native-base/lib/typescript/theme/base/colors';
import { FeatherIcon } from '../../@types';
import { Icon } from '../Icon';

type IconButtonProps = {
  testID?: string;
  icon: FeatherIcon;
  color?: IColors;
  onPress: () => void;
  size?: number;
};

export const IconButton = (props: IconButtonProps) => {
  const { icon, onPress, color = 'muted.400', size = 6, testID } = props;

  return (
    <Pressable onPress={onPress} testID={testID}>
      <Icon size={size} color={color} name={icon} />
    </Pressable>
  );
};
