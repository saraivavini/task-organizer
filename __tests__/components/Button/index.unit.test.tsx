import { Button } from '../../../src/components';
import { componentBuilder } from '../../utils/componentBuilder';

const { renderComponent } = componentBuilder(Button, {
  label: '',
  onPress: jest.fn(),
});

describe('Button tests', () => {
  it('Should renders correctly', () => {
    const { component } = renderComponent();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
