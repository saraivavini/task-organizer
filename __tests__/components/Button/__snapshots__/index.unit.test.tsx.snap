// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Should renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="brand"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0c59c3",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
    testID="button"
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    />
  </View>
</RNCSafeAreaProvider>
`;
